MapleServer2/Packets/GuildPacket.cs

```diff
@@ -0,0 +1,328 @@
+using MapleServer2.Types;
+
+namespace MapleServer2.Packets
+{
+    public static class GuildPacket
+    {
+        private enum GuildPacketMode : byte
+        {
+            UpdateGuild = 0x0,
+            Create = 0x1,
+            Invite = 0x4C,
+            CheckInConfirm = 0xF,
+            Create2 = 0x12,
+            GuildNoticeChange = 0x1A,
+            Create3 = 0x20,
+            Unk1 = 0x24,
+            UpdateGuildExp = 0x31,
+            UpdateGuildFunds = 0x32,
+            UpdatePlayerContribution = 0x33,
+            GuildNoticeConfirm = 0x3E,
+            GuildWindow = 0x54,
+            List = 0x5A,
+            UpdateGuildFunds2 = 0x59, //unk
+            UpdatePlayerDonation = 0x6E,
+        }
+
+        public static Packet Invite(Player player, string guildName)
+        {
+            PacketWriter pWriter = PacketWriter.Of(SendOp.GUILD);
+            pWriter.WriteByte((byte) GuildPacketMode.Invite);
+            pWriter.WriteUnicodeString(player.Name);
+            pWriter.WriteUnicodeString(guildName);
+            return pWriter;
+        }
+
+        public static Packet Create(string guildName)
+        {
+            PacketWriter pWriter = PacketWriter.Of(SendOp.GUILD);
+            pWriter.WriteByte((byte) GuildPacketMode.Create);
+            pWriter.WriteByte(0x0);
+            pWriter.WriteUnicodeString(guildName);
+            return pWriter;
+        }
+
+        public static Packet UpdateGuild(GameSession session, string guildName)
+        {
+            PacketWriter pWriter = PacketWriter.Of(SendOp.GUILD);
+            pWriter.WriteByte(0x0);
+            pWriter.WriteLong(1); // guild ID
+            pWriter.WriteUnicodeString(guildName);
+            pWriter.WriteUnicodeString(""); // guildMark Url
+            pWriter.WriteByte(0x3C); // guildC.apacity
+            pWriter.WriteUnicodeString("");
+            pWriter.WriteUnicodeString(""); // guild.Notice
+            pWriter.WriteLong(session.Player.AccountId); // guild.Leader.AccountId
+            pWriter.WriteLong(session.Player.CharacterId); // guild.Leader.CharacterId
+            pWriter.WriteUnicodeString(session.Player.Name); // guild.Leader.Player.Name
+            pWriter.WriteLong(1609108139); // guild creation timestamp
+            pWriter.WriteByte(0x0);
+            pWriter.WriteInt(1000);
+            pWriter.WriteInt(0);
+            pWriter.WriteInt(0);
+            pWriter.WriteInt(0);
+            pWriter.WriteInt(0);
+            pWriter.WriteInt(0);
+            pWriter.WriteInt(0);
+            pWriter.WriteInt(0);
+            pWriter.WriteByte(0x1);
+            pWriter.WriteInt(0);
+            pWriter.WriteInt(0); // guild.Exp
+            pWriter.WriteInt(0); // guild.Funds
+            pWriter.WriteByte(0x0);
+            pWriter.WriteInt(0);
+            pWriter.WriteByte(0x0); // guild.Members.Count and also loop for each member ?
+            pWriter.WriteByte(0x6); // loop for guild ranks ?
```

Comment:
I think this should be in a loop, maybe next change though. You can store the ranks in a list or an array.